### PR TITLE
take the size of 'shared_secret' itself rather than its address

### PR DESCRIPTION
**Pull request by yashbhutwala**

### CODE REVIEW

File: lightningd/peer_htlcs.c [hunk 1]
```diff
@@ -1392,7 +1392,7 @@ static bool channel_added_their_htlc(struct channel *channel,
 
 	/* FIXME: Our wire generator can't handle optional elems in arrays,
 	 * so we translate all-zero-shared-secret to NULL. */
-	if (memeqzero(shared_secret, sizeof(&shared_secret)))
+	if (memeqzero(shared_secret, sizeof(*&shared_secret)))
```
Comment (niftynei) [inline lightningd/peer_htlcs.c]: why not just `sizeof(shared_secret)`?

Comment (yashbhutwala) [reply lightningd/peer_htlcs.c]: To be a fancy 😺.. fixed 😄

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3138/commits/16bf1662dee28d91b2b7be72933a8269ccb9c483

File: lightningd/peer_htlcs.c [hunk 2]
```diff
@@ -1392,7 +1392,7 @@ static bool channel_added_their_htlc(struct channel *channel,
 
 	/* FIXME: Our wire generator can't handle optional elems in arrays,
 	 * so we translate all-zero-shared-secret to NULL. */
-	if (memeqzero(shared_secret, sizeof(&shared_secret)))
+	if (memeqzero(shared_secret, sizeof(shared_secret)))
```
Comment (rustyrussell) [inline lightningd/peer_htlcs.c]: You mean `sizeof(*shared_secret)` here!

**Pull request merged**